File: Examples/CSharp/ModifyingAndConvertingImages/AlignHorizontalAndVeticalResolutionsOfImage.cs
using System;
using Aspose.Imaging.FileFormats.Tiff;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API references 
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class AlignHorizontalAndVeticalResolutionsOfImage
    {
        public static void Run()
        {
            Console.WriteLine("Running example AlignHorizontalAndVeticalResolutionsOfImage");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load an image and convert the image instance to TiffImage.
            using (TiffImage image = (TiffImage)Image.Load(dataDir + "SampleTiff1.tiff"))
            {
                // Call the align resolution method and save the results to the output path.
                image.AlignResolutions();
                image.Save(dataDir + "AlignHorizontalAndVeticalResolutionsOfImage_out.tiff");
                int framesCount = image.Frames.Length;
                for (int i = 0; i < framesCount; i++)
                {
                    TiffFrame frame = image.Frames[i];
                    // All resolutions after alignment must be equal.
                    Console.WriteLine(
                        "Horizontal and vertical resolutions are equal:" +
                        ((int)frame.VerticalResolution == (int)frame.HorizontalResolution));
                }
            }

            Console.WriteLine("Finished example AlignHorizontalAndVeticalResolutionsOfImage");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/AlignHorizontalAndVeticalResolutionsOfImage.vb
Imports Aspose.Imaging.FileFormats.Tiff

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class AlignHorizontalAndVeticalResolutionsOfImage
        Public Shared Sub Run()
            ' To get proper output please apply a valid Aspose.Imaging License. You can purchase full license or get 30 day temporary license from http:// Www.aspose.com/purchase/default.aspx.")
            ' ExStart:AlignHorizontalAndVeticalResolutionsOfImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load an image and convert the image instance to TiffImage
            Using image__1 As TiffImage = DirectCast(Image.Load(dataDir & Convert.ToString("SampleTiff1.tiff")), TiffImage)
                ' Call the align resolution method
                image__1.AlignResolutions()

                ' Save the results to output path.
                image__1.Save(dataDir & Convert.ToString("AlignHorizontalAndVeticalResolutionsOfImage_out.tiff"))

                Dim framesCount As Integer = image__1.Frames.Length
                For i As Integer = 0 To framesCount - 1
                    Dim frame As TiffFrame = image__1.Frames(i)
                    ' All resolutions after aligment must be equal

                    Dim verticalResolution = frame.VerticalResolution
                    Dim horizontalResolution = frame.HorizontalResolution

                    Console.WriteLine("Horizontal and vertical resolutions are equal:" & verticalResolution & ":" & horizontalResolution)
                Next
            End Using
            ' ExEnd:AlignHorizontalAndVeticalResolutionsOfImage
        End Sub
    End Class
End Namespace